### PR TITLE
Avoid analyzing classes in Manifest methods while checking ReClassesHaveNoReferencesCondition

### PR DESCRIPTION
**Pull request by hernanmd**
This PR fixes problems described in #16953.

Update the `violators` method in ReClassesHaveNoReferencesCondition.
Add isManifest method for RB classes.
Add testIsManifest.
Fix failing #testTransform method in RBRemoveProtocolTransformationTest

### CODE REVIEW

**Review by balsa-sarenac — APPROVED** [review]

File: src/Refactoring-Core/RBClass.class.st [hunk 1]
```diff
@@ -160,6 +160,14 @@ RBClass >> isEmptyClass [
 					  self selectors isEmpty]]
 ]
 
+{ #category : 'testing' }
+RBClass >> isManifest [ 
+
```
Comment (Ducasse) [inline src/Refactoring-Core/RBClass.class.st:165]: Can you add some tests for this because it gives me shivers :)
In particular because reading it is cumbersome (how superclass can work in none realClass)
I know how but reading it is strange.

Comment (hernanmd) [reply src/Refactoring-Core/RBClass.class.st:165]: Thanks Stef; I updated the isManifest implementation, so now it doesn't check the realClass. I also included a new test when the superclass is nil, and the implementation handles this case.

**Review by jecisc — APPROVED** [review]
